zebra-test/src/command.rs

```diff
@@ -17,23 +17,29 @@ pub fn test_cmd(command_path: &str, tempdir: &PathBuf) -> Result<Command> {
     Ok(cmd)
 }
 
-/// Create a temp directory and optional config file
-pub fn tempdir(create_config: bool) -> Result<(PathBuf, impl Drop)> {
+/// Create a temp directory and optional default config file with optional additional options
+pub fn tempdir(
+    create_config: bool,
+    additional_config: Option<String>,
+) -> Result<(PathBuf, impl Drop)> {
```

Comment:
because these are all conditional flags I feel like this would be a good fit for a builder pattern.

Reply:
Can we add the network config to the default testing config?

The network config is like the sled path config. It is needed by every test that opens a zcashd listener, but doesn't change the behaviour of other tests.

Here's another reason to change the default testing config:

It's easy to add tests which need the config, but don't have it. Because we will only get failures when we add multiple tests *without* the config. And even then, the failures can be intermittent.

Reply:
(If future tests need incompatible configs, we can implement a builder pattern.)

Reply:
Isn't the mainnet the default network (https://github.com/ZcashFoundation/zebra/blob/main/zebra-chain/src/parameters/network.rs#L16) ? If we add just mainnet as the network into the default config we dont win anything as far as i can see and we will not be able to change it from the tests. With what we have now the `additional_config` can be used to change to testnet with something like:

`[network]\nnetwork = 'Testnet'\n`

Maybe i am not understanding correctly the request.

Reply:
If we add the listen_addr to all configs, we don't need this change, which would also resolve the issue:
https://github.com/ZcashFoundation/zebra/pull/920/files#r473304292